src/internal_events/exec.rs

```diff
@@ -38,49 +44,75 @@ impl InternalEvent for ExecEventsReceived<'_> {
 }
 
 #[derive(Debug)]
-pub struct ExecFailed<'a> {
+pub struct ExecFailedError<'a> {
     pub command: &'a str,
     pub error: std::io::Error,
 }
 
-impl InternalEvent for ExecFailed<'_> {
+impl InternalEvent for ExecFailedError<'_> {
     fn emit_logs(&self) {
         error!(
             message = "Unable to exec.",
             command = %self.command,
             error = ?self.error,
+            error_type = "failed",
+            stage = "receiving",
         );
     }
 
     fn emit_metrics(&self) {
+        counter!(
+            "component_errors_total", 1,
+            "command" => self.command.to_owned(),
+            "error" => self.error.to_string(),
+            "error_type" => "command_failed",
+            "stage" => "receiving",
+        );
+        // deprecated
         counter!(
             "processing_errors_total", 1,
             "command" => self.command.to_owned(),
-            "error_type" => "failed",
+            "error" => self.error.to_string(),
+            "error_type" => "command_failed",
+            "stage" => "receiving",
         );
     }
 }
 
 #[derive(Debug)]
-pub struct ExecTimeout<'a> {
+pub struct ExecTimeoutError<'a> {
     pub command: &'a str,
     pub elapsed_seconds: u64,
+    pub error: String,
```

Comment:
Is it worth using a string slice here like for `command`?

Reply:
I think I'd prefer replacing `String` with the real error type 🤔